client/state/themes/selectors.js

```diff
@@ -602,6 +602,23 @@ export function isPremiumThemeAvailable( state, themeId, siteId ) {
 }
 
 /**
+ * Whether a given theme is installed or can be installed on a Jetpack site.
+ *
+ * @param  {Object}  state   Global state tree
+ * @param  {String}  themeId Theme ID for which we check availability
+ * @param  {Number}  siteId  Site ID
+ * @return {Boolean}         True if siteId is a Jetpack site on which theme is installed or can be installed.
+ */
+export function isThemeAvailableOnJetpackSite( state, themeId, siteId ) {
```

Comment:
I'm trying to think why we can't get away with just the capability checks here. Apart from premiums, _any other theme_ should be available for a Jetpack site: free wpcom, custom, wporg. What is the scenario where one of those types should return false here?